bound/kt/src/test/kotlin/web5/sdk/crypto/keys/InMemoryKeyManagerTest.kt

```diff
@@ -1,47 +1,112 @@
 package web5.sdk.crypto.keys
 
-import org.junit.jupiter.api.Assertions.assertThrows
-import org.junit.jupiter.api.Assertions.assertNotNull
-
-import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.*
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.fail
+import web5.sdk.UnitTestSuite
+import web5.sdk.crypto.Ed25519Generator
 import web5.sdk.rust.Web5Exception
 
-import web5.sdk.rust.ed25519GeneratorGenerate as rustCoreEd25519GeneratorGenerate
-
 class InMemoryKeyManagerTest {
+  @Nested
+  @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+  inner class ImportPrivateJwk {
+    private val testSuite = UnitTestSuite("in_memory_key_manager_import_private_jwk")
 
-  @Test
-  fun `test key manager`() {
-    val privateJwk = rustCoreEd25519GeneratorGenerate()
+    @AfterAll
+    fun verifyAllTestsIncluded() {
+      if (testSuite.tests.isNotEmpty()) {
+        println("The following tests were not included or executed:")
+        testSuite.tests.forEach { println(it) }
+        fail("Not all tests were executed! ${this.testSuite.tests}")
+      }
+    }
 
-    val keyManager = InMemoryKeyManager(listOf(Jwk.fromRustCoreJwkData(privateJwk)))
+    @Test
+    fun test_must_be_private_jwk() {
+      this.testSuite.include()
 
-    val signer = keyManager.getSigner(Jwk.fromRustCoreJwkData(privateJwk))
-    val payload = signer.sign("abc".toByteArray())
+      val keyManager = InMemoryKeyManager(listOf())
+      val privateJwk = Ed25519Generator.generate()
+      val publicJwk = privateJwk.copy(d = null)
 
-    assertNotNull(payload)
-  }
+      val exception = assertThrows<Web5Exception.Exception> {
+        keyManager.importPrivateJwk(publicJwk)
+      }
+
+      assertEquals("parameter error private_jwk must be a private key", exception.msg)
+      assertEquals("Parameter", exception.variant)
+    }
+
+    @Test
+    fun test_successfully_imports_and_returns_public_jwk() {
+      this.testSuite.include()
```

Comment:
nit: i think it's idiomatic to omit `this` in kotlin where possible

Reply:
done